internal_worker.go

```diff
@@ -359,7 +360,7 @@ type hostEnvImpl struct {
 	workflowAliasMap                 map[string]string
 	activityFuncMap                  map[string]activity
 	activityAliasMap                 map[string]string
-	encoding                         gobEncoding
+	encoding                         jsonEncoding
```

Comment:
shouldn't this be of encoding type?

Reply:
yes, it should be using the interface encoding here and then assign a concrete implementation to it.